### DISCUSSION TITLE
Log Rotation attributes

### DISCUSSION THREAD

**Discussion by cubukinas** [rabbitmq/chef-cookbook#601] [Q&A]
I've tried changing the default attributes for logrotate in my recipe or the attributes
```bash
node.default['rabbitmq']['logrotate']['rotate'] = 5
node.default['rabbitmq']['logrotate']['frequency'] = 'daily'
```
But to avail success logrotate is still set as default in my test environment, we're including the default recipe from upstream.
Other attributes such as log_level work fine.
<details>
  <summary>/etc/logrotate.conf</summary>  

  ```bash
  # see "man logrotate" for details
# rotate log files weekly
weekly

# keep 4 weeks worth of backlogs
rotate 4

# create new (empty) log files after rotating old ones
create

# use date as a suffix of the rotated file
dateext

# uncomment this if you want your log files compressed
#compress

# RPM packages drop log rotation information into this directory
include /etc/logrotate.d

# no packages own wtmp and btmp -- we'll rotate them here
/var/log/wtmp {
    monthly
    create 0664 root utmp
	minsize 1M
    rotate 1
}

/var/log/btmp {
    missingok
    monthly
    create 0600 root utmp
    rotate 1
}
  ```
  
</details>
<details>
  <summary>/etc/logrotate.d/rabbitmq-server</summary>  

  ```bash
/var/log/rabbitmq/*.log {
        weekly
        missingok
        rotate 20
        compress
        notifempty
}
  ```
</details>
What could be the issue here why is the file not being modified?

**Comment by cubukinas** [comment]
The file  /etc/logrotate.d/rabbitmq-server is not even being modified by chef then what is being modified as I checked in my nodes after running the upstream default config the modified date stays the same.
What is being changed here?

**Comment by fozboz** [comment]
I would guess that you are hitting #550

**Comment by cubukinas** [comment]
So there's basically no way to manage logs via upstream cookbook configuration for newer rabbitmq versions as logging levels do not work and new attributes for the built in log rotation have not been implemented such as keeping logs by file-size such as 
`log.file.rotation.size = 10485760`?

**Reply by lukebakken** [reply]
The chef cookbook appears to be a bit out-of-date with what RabbitMQ supports. @michaelklishin thoughts about giving this more attention? I don't know if this cookbook is mostly community maintained or RabbitMQ team maintained.

@cubukinas to be honest the RabbitMQ team's focus has been on Kubernetes and the cluster operator for quite some time now - https://www.rabbitmq.com/kubernetes/operator/operator-overview.html
